docs/introduction-to-smart-contracts.rst

```diff
@@ -179,36 +178,33 @@ web3.js or a similar module)::
         }
     })
 
-Note how the automatically generated function ``balances`` is called from
-the user interface.
-
 .. index:: coin
 
-The constructor is a special function which is run during creation of the contract and
+The :ref:`constructor<constructor>` is a special function run during the creation of the contract and
 cannot be called afterwards. It permanently stores the address of the person creating the
-contract: ``msg`` (together with ``tx`` and ``block``) is a special global variable that
-contains some properties which allow access to the blockchain. ``msg.sender`` is
+contract. The ``msg`` variable (together with ``tx`` and ``block``) is a
+:ref:`special global variable <special-variables-functions>` that
+contains properties which allow access to the blockchain. ``msg.sender`` is
 always the address where the current (external) function call came from.
 
-Finally, the functions that will actually end up with the contract and can be called
-by users and contracts alike are ``mint`` and ``send``.
-If ``mint`` is called by anyone except the account that created the contract,
-nothing will happen. This is ensured by the special function ``require`` which
-causes all changes to be reverted if its argument evaluates to false.
-The second call to ``require`` ensures that there will not be too many coins,
-which could cause overflow errors later.
+The functions that make up the contract, and that users and contracts can call are ``mint`` and ``send``.
+
+The ``mint`` function sends an amount of newly created coins to another address.
+The :ref:`require <assert-and-require>` function defines conditions that throw an exception if not met, and reverts all changes. In
```

Comment:
Also, the function does not define the condition, the function call does it. But maybe that is pedantic...